services/src/main/java/org/keycloak/services/resources/KeycloakApplication.java

```diff
@@ -145,29 +151,34 @@ public void run(KeycloakSession session) {
                 // TODO up here ^^
 
                 ApplianceBootstrap applianceBootstrap = new ApplianceBootstrap(session);
-                exportImportManager[0] = new ExportImportManager(session);
-
-                boolean createMasterRealm = applianceBootstrap.isNewInstall();
-                if (exportImportManager[0].isRunImport() && exportImportManager[0].isImportMasterIncluded()) {
-                    createMasterRealm = false;
-                }
-
-                if (createMasterRealm) {
-                    applianceBootstrap.createMasterRealm();
+                bootstrapState.exportImportManager = new ExportImportManager(session);
+                bootstrapState.newInstall = applianceBootstrap.isNewInstall();
+                if (bootstrapState.newInstall) {
+                    if (!(bootstrapState.exportImportManager.isRunImport() && bootstrapState.exportImportManager.isImportMasterIncluded()) 
+                        && !(getImportDirectory().filter(bootstrapState.exportImportManager::isImportMasterIncludedAtStartup).isPresent())) {
+                        applianceBootstrap.createMasterRealm();
+                    }
```

Comment:
nit: Would it be possible to replace conditions with named boolean vars to clearly see what is happening there? It'd be better for readability.

Reply:
Sure, added something similar to what it used to be.